### ISSUE TITLE
Blocking submission for second listed version in 6 steps

### ISSUE THREAD

**Issue by ValentinaPC** [mozilla/addons-server#4420]
Steps to reproduce:
1. Submit an unlisted add-on https://addons.allizom.org/en-US/developers/addon/submit/upload-unlisted
2. Click "New Version" link for the submitted add-on and select "On this site" option
3. Select a valid file
4. _Refresh_ the page and click "Continue" (_button still active even if no file appear to be selected after page refresh_)
5. Click `back`
6. Select a valid file again

Expected results:
The file selected at step 8 is validated and can be added as a second listed version for this add-on.

Actual results:
After step 8 the message "There was a problem contacting the server" is displayed.

Notes/Issues:
Reproducing only if the first submitted version is unlisted.
Verified on FF50(Win 7). Issue is reproducing on AMO-stage and -dev.
Video for this issue: 
![error](https://cloud.githubusercontent.com/assets/15685960/22112233/d7dcd788-de6a-11e6-8614-c85ca4e2385d.gif)

**Comment by eviljeff** [comment]
I think what's happening here is the upload xhr is being captured and redirected to the details step, because the metadata wasn't filled in at 5.  (The details page not being the correct destination for a upload so it fails)

**Comment by ValentinaPC** [comment]
Verified on AMO-dev FF50(Win 7).
After following the exact steps as explained in the first comment, the message "Version `X` already exists, or was uploaded before", while selecting again the same file, for the second version.
@eviljeff Is this the fix? (from my point of view seems legit, even if, the submission did not got to an end)
Posfix video:
![submiterr](https://cloud.githubusercontent.com/assets/15685960/22287694/715bc12c-e2fc-11e6-890e-297cd788fe85.gif)

**Comment by eviljeff** [comment]
It's correct for the current behaviour - you're submitting the same version twice which isn't allowed.  But we should handle it better than a tiny error message in red.  Can you log a follow-up issue to handle duplicate versions better?

**Comment by ValentinaPC** [comment]
Marking this verified-fixed.
